reactioncommerce/reaction
# Enable discounts for single shop checkout when using a multi-shop enabled payment provider (e.g Stripe)

Resolves #3005 

**Added / Changed:** 
- Updated `getTotalByShop()` to subtract discount from shopTotal (if only one shop is participating in the order)
- Updated the `discounts/codes/apply` method to check that discounts are only applied to single shop carts
- Updated the Add Discount UI to show error alert to user when trying to add discount on a multi-shop checkout
- Added two extra tests for discounts/codes/apply method


## How To Test:
- Create a discount code (under Dashboard > Payment)
- While still under Payment, enable Stripe as payment provider
- Add an item to cart and proceed to checkout. Apply the discount code created earlier, then add card details and pay.
- Confirm that the discount is applied to the order on both the "Thank you page" and in the Order Dashboard

Second test
- Add two items from separate shops into a cart. (you'll need to make a second shop for this)
- Proceed to checkout and try to apply a discount code
- Confirm that the code is not applied and an error message to explain why is shown
